pkg/microservice/reaper/core/service/reaper/artifact.go

```diff
@@ -87,26 +88,36 @@ func artifactsUpload(ctx *meta.Context, activeWorkspace string, artifactPaths []
 	}
 
 	artifactPath := filepath.Join(activeWorkspace, artifactPaths[0])
-	temp, err := ioutil.TempFile("", "*artifact.tar.gz")
-	if err != nil {
-		log.Errorf("failed to create temp file %s", err)
-		return err
-	}
+	isDir := fs.IsDir(artifactPath)
+	if isDir {
+		temp, err := ioutil.TempFile("", "*artifact.tar.gz")
```

Comment:
ioutil is deprecated, use os.CreateTemp instead
you can also use `util.GenerateTmpFile()`